### DISCUSSION TITLE
Targeting Unity on Apple Silicon

### DISCUSSION THREAD

**Discussion by edsonedge** [Wasted-Audio/hvcc#228] [Q&A]
On the unity page it says : " Heavy can generate 64-bit OSX bundles"... does that mean it won't compile correctly on M1,M2,M3 Silicone Mac's ? Specifically I am thinking about switching from libPdIntegration in Unity to hvcc, since I am having trouble with multiple instances of LibPdInstance in Unity on Mac M2 (works fine on Windows machines)

**Comment by dromer** [comment]
It should totally build, but we need someone to give it a shot.

Personally I have no experience with the Unity generator and I don't have any Apple Silicon. So this makes testing this particular integration quite difficult of course.

The documentation for Unity is definitely outdated and suggestions and pull requests for updating it are appreciated.

If you give this a try and run into issues please report them as tickets in the Issue tracker so that someone can look at fixing them :)

**Reply by edsonedge** [reply]
Thanks... I'll give it a shot.

**Comment by edsonedge** [comment]
Nah, I can't get it to work in Unity. It compiles and builds fine on my M2 machine, but when I load the files in unity and start it up, I get this error:

DllNotFoundException: Hv_pdTest_AudioLib assembly: type: member:(null)
Hv_pdTest_Context..ctor (System.Double sampleRate, System.Int32 poolKb, System.Int32 inQueueKb, System.Int32 outQueueKb) (at Assets/hvcc-Mac/Hv_pdTest_AudioLib.cs:341)
Hv_pdTest_AudioLib.Awake () (at Assets/hvcc-Mac/Hv_pdTest_AudioLib.cs:220)

Strange that it says Dll not found, I would have thought it should be "bundle" not found, but might just be a Unity thing.

**Comment by edsonedge** [comment]
Strike that... it is doable on M1/M2/M3 machines but only if you have an Intel version of Unity (i.e. you can download either a silicon or an intel version of Unity) 

So it works, but only through Rosetta (intel emulator) which means it is not using the power of M machines to it's fullest ability.

All the best

**Reply by dromer** [reply]
It means that someone will need to adjust the current templates to work for Apple ARM architecture.

**Reply by dromer** [reply]
@edsonedge please test the latest develop branch, it has been modified to use cmake and should build!
